tests/test_stubs.py

```diff
@@ -426,6 +426,44 @@ def test_update_class(self):
         expected = Signature(parameters=[Parameter('cls', Parameter.POSITIONAL_OR_KEYWORD)])
         assert sig == expected
 
+    def test_update_arg_ignore_existing_anno(self):
+        """Update stubs only bases on traces."""
+        sig = Signature.from_callable(UpdateSignatureHelper.has_annos)
+        ignore_existing_annotations = True
+        sig = update_signature_args(sig, {'a': str, 'b': bool}, False, ignore_existing_annotations)
+        params = [
+            Parameter('a', Parameter.POSITIONAL_OR_KEYWORD, annotation=str),
+            Parameter('b', Parameter.POSITIONAL_OR_KEYWORD, annotation=bool),
+        ]
+        assert sig == Signature(parameters=params, return_annotation=int)
+
+    def test_update_self_ignore_existing_anno(self):
+        """Don't annotate first arg of instance methods with ignore_existing_annotations"""
+        sig = Signature.from_callable(UpdateSignatureHelper.an_instance_method)
+        sig = update_signature_args(sig, {'self': UpdateSignatureHelper}, True, True)
+        expected = Signature(parameters=[Parameter('self', Parameter.POSITIONAL_OR_KEYWORD)])
+        assert sig == expected
+
+    def test_update_arg_ignore_existing_anno_NoneType(self):
+        """Update arg annotations from types"""
+        sig = Signature.from_callable(UpdateSignatureHelper.has_annos)
+        sig = update_signature_args(sig, {'a': NoneType, 'b': int}, False, True)
+        params = [
+            Parameter('a', Parameter.POSITIONAL_OR_KEYWORD, annotation=inspect.Parameter.empty),
```

Comment:
I think this test is wrong. `annotation=None` would be correct here.

Reply:
Related to the discussion in checking `None` and `NoneType` TODO

Reply:
Yes, `annotation=None` is what we want here, not `annotation=inspect.Parameter.empty`. I think this test can just be removed from this PR because  handling of `NoneType` is not related to handling of `ignore_existing_annotations`. The PR that fixes #5 can add a test similar to this (but doesn't need to use `ignore_existing_annotations`).